packages/tools/viewer-alpha/src/viewerFactory.ts

```diff
@@ -31,22 +40,29 @@ export async function createViewerForCanvas(canvas: HTMLCanvasElement, options?:
 
     // Create an engine instance.
     let engine: AbstractEngine;
-    switch (finalOptions.engine) {
-        case undefined:
+    switch (finalOptions.engine ?? getDefaultEngine()) {
         case "WebGL": {
             // eslint-disable-next-line @typescript-eslint/naming-convention, no-case-declarations
             const { Engine } = await import("core/Engines/engine");
             engine = new Engine(canvas, undefined, options);
             break;
         }
-        // case "WebGPU": {
-        //     // eslint-disable-next-line @typescript-eslint/naming-convention
-        //     const { WebGPUEngine } = await import("core/Engines/webgpuEngine");
-        //     const webGPUEngine = new WebGPUEngine(canvas, options);
-        //     await webGPUEngine.initAsync();
-        //     engine = webGPUEngine;
-        //     break;
-        // }
+        case "WebGPU": {
+            const [webgpuModule] = await Promise.all([
+                import("core/Engines/webgpuEngine"),
+                import("core/Engines/WebGPU/Extensions/engine.alpha"),
+                import("core/Engines/WebGPU/Extensions/engine.rawTexture"),
+                import("core/Engines/WebGPU/Extensions/engine.cubeTexture"),
+                import("core/Engines/WebGPU/Extensions/engine.renderTarget"),
+                import("core/Engines/WebGPU/Extensions/engine.renderTargetCube"),

```

Comment:
Presumably all these should happen automatically in the right places inside @babylonjs/core.

Reply:
Cc @sebavan ri get his thoughts

Reply:
Oh I see we only import the Engine (WebGL) version... Ok that will be an interesting change! I'll take care of it!